src/librustc_trans/debuginfo/metadata.rs

```diff
@@ -657,6 +657,15 @@ pub fn type_metadata<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
     metadata
 }
 
+fn remap_path(sess: &Session, path: &str) -> String {
+    for (old, new) in sess.opts.debug_prefix_map.clone() {
+        if path.starts_with(&old) {
+            return new + &path[old.len()..];
+        }
```

Comment:
It seems unnecessary to clone the whole map first; it would be better to iterate references, then only clone the "new" once we've found a match.